micropub.php

```diff
@@ -481,6 +496,37 @@ private static function handle_authorize_error( $code, $msg ) {
 		static::respond( $code, $msg );
 	}
 
+	private static function post_status( $mf2 ) {
+		$props = $mf2['properties'];
+		if ( isset( $props['visibility'] ) ) {
+			if ( array( 'private' ) === $props['visibility'] ) {
+				return 'private';
+			}
+			// Public or Private are the only options
+			if ( array( 'public' ) !== $props['visiblity'] ) {
+				return null;
+			}
+		}
+		if ( ! isset( $props['post-status'] ) ) {
+			return get_option( 'micropub_default_post_status' );
+		} else {
+			//  According to the proposed specification these are the only two properties supported.
+			// https://indieweb.org/Micropub-extensions#Post_Status
+			// For now these are the only two we will support even though WordPress defaults to 8 and allows custom
+			// But makes it easy to change
+			$statuslist = array( array( 'published' ), array( 'draft' ) );
+			if ( ! in_array( $props['post-status'], $statuslist, true ) ) {
+				return null;
+			}
+			// Map published to the WordPress property publish.
+			if ( array( 'published' ) === $props['post-status'] ) {
+				return 'publish';
+			}
+			return $props['post-status'];
```

Comment:
minor nit: probably change this to `return 'draft';` for consistency and readability, since we return `'publish'` as a string above, not an array, and for readability since it's guaranteed to be `'draft'` here.

Reply:
Again, unless we support additional post-status properties in future. Is it better to make it so to add properties, we just add them to the whitelist above, or wait till we need it?

Reply:
good point! predicting the future is difficult and fraught, though, so i lean toward only designing for the present, which is a known quantity. let's go with `return 'draft';`.